src/session.ts

```diff
@@ -261,15 +247,19 @@ function getMiddlewareAuthPathRegex(pathGlob: string) {
 
 async function redirectToSignIn() {
   const headersList = await headers();
-  const url = headersList.get('x-url') ?? '';
+  const url = headersList.get('x-url');
+
+  if (!url) {
```

Comment:
This would already throw an error because if `url` is undefined then it would throw when we attempt `new URL(url)` a few lines down. I added this to be more descriptive over the generic "Invalid URL" error we'd otherwise get.